src/common.rs

```diff
@@ -124,3 +124,69 @@ pub fn get_widgets() -> color_eyre::eyre::Result<
     }
     Ok(widgets)
 }
+
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+pub struct PredecessorIdFunctionArgs {
+    predecessor_id: near_primitives::types::AccountId,
+    key: String,
+}
+
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+pub struct PublicKeyFunctionArgs {
+    public_key: near_crypto::PublicKey,
+    key: String,
+}
+
+pub async fn is_write_permission_granted(
+    config: near_cli_rs::config::Config,
+    network_config: near_cli_rs::network_for_transaction::NetworkForTransactionArgs,
```

Comment:
Instead of passing two arguments, pass just a single one, and you can pass it by reference as you don't plan to change it:

```suggestion
    network_config: &near_cli_rs::config::NetworkConfig,
```